velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -1019,14 +1036,21 @@ bool assertEqualResults(
 
 bool assertEqualResults(
     const MaterializedRowMultiset& expectedRows,
+    const TypePtr& expectedType,
     const std::vector<RowVectorPtr>& actual) {
   MaterializedRowMultiset actualRows;
   for (auto vector : actual) {
     auto rows = materialize(vector);
     std::copy(
         rows.begin(), rows.end(), std::inserter(actualRows, actualRows.end()));
   }
-  return assertEqualResults(expectedRows, actualRows, "Unexpected results");
+  // Get the expectedType from actual as the expectedRows can be 0
+  const auto expType = (expectedType == nullptr)
+      ? actual.at(0)->childAt(0)->type()
+      : expectedType;
+  const auto actType = actual.at(0)->childAt(0)->type();
```

Comment:
This should be nullptr if actualRows is empty.

Reply:
Fixed.